src/inputs/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { render, fireEvent } from '@testing-library/preact';
+import TextField from './TextField';
+import { h } from 'preact'
+import styles from './TextField.module.css'
+
+describe('TextField', () => {
+  const props = {
+    onChange: jest.fn(),
+    label: 'test',
+    name: 'test-name',
+    placeholder: 'test-placeholder',
+    id: 'test-id',
+  };
+
+  it('should call onChange handler when typing onto it', () => {
+    const { getByPlaceholderText } = render(<TextField {...props} />);
+    const input = getByPlaceholderText('test-placeholder');
+    fireEvent.change(input, { target: { value: 'new value' } });
+    expect(props.onChange).toHaveBeenCalled();
```

Comment:
**thought:** I have no idea why it does not work either. Probably it can relate to `preact` library that you still use, and which we already migrated from. Hubert has almost the same implementation (but in React), and it works for him. For now, I suggest a little workaround to not be stucked here. Test if changing value changes it's value in input.

```

  it('should call change input value when typing onto it', () => {
    const { getByPlaceholderText } = render(<TextField {...props} />);
    const input = getByPlaceholderText('test-placeholder');
    fireEvent.change(input, { target: { value: 'new value' } });
    expect(input.value).toBe('new value');
  });
```